data/tutorials/language/1ms_01_functors.md

```diff
@@ -8,183 +8,381 @@ category: "Module System"
 
 # Functors
 
-Functors are probably one of the most complex features of OCaml, but you don't
-have to use them extensively to be a successful OCaml programmer. Actually,
-you may never have to define a functor yourself, but you will surely encounter
-them in the standard library. They are the only way of using the Set and Map
-modules, but using them is not so difficult.
+## Introduction
 
-## What Are Functors and Why Do We Need Them?
+Learning goals:
+- How to use a functor
+- How to write a functor
+- When to use a functor, some cases
 
-A functor is a module that is parametrised by another module, just like a
-function is a value which is parametrised by other values, the arguments.
+A functor is just a parametrized module.
```

Comment:
I think the older description was better here since it describes a functor by analogy with something else that people may be familiar with. 

I'd take it a little further and say that functors _are functions_ but that instead of regular values they take modules as inputs and create modules as outputs. 

This intuition I think can help with other things, like the "transitive closures leading to modules" prerequisite.

Reply:
Sorry, I should have said what I've written here is interim text. The original intro had some good ideas but it needs to be rewritten. This also applies to the transitive closure thing, of course we're not going to publish this as is.

Reply:
No worries, and nothing to be sorry for! 🙏🏼

Reply:
Should we also say something like:
> The term _functor_ comes from a branch of mathematics called category theory. Although there is a loose relation between OCaml functors and category theory functors, no knowledge of the latter is required to understand and work with the former.